src/lib/peer.js

```diff
@@ -79,9 +91,11 @@ function Peer(config) {
 
     function router(envelope) {
         var nextHop = self.fingerTable.bestCandidate(envelope.dstId);
-        log('nextHop:', nextHop, envelope);
-        if (nextHop === self.peerId.toHex()) {
+        console.log('nextHop:', nextHop, envelope);
+        if (nextHop === self.peerId.toHex() && envelope.dstId === self.peerId.toHex()) {
```

Comment:
Imagine a setup of 3 nodes, with IDs peer1, peer2, peer3.
Now, what happens if peer1 wants to send a message to peer4, which obviously is not connected?
There is no way for peer1 to tell if peer4 is connected right now or not, therefore he looks for the best node (the nextHop) to send the message to. If, by coincident, the next best hop again is the origin peer (peer1), we end up with a raised message event, containing a message for peer4.
Of course, you could also check in the message handler if the destination matches my peer id. But my understanding of the message event is to get only raised for messages where the final destination has been reached.